Reworks janitor cyborg cleaning, focus on the slipping

LemonInTheDark opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Alt of #64105 and #64126 (I'm sorry Novva, I should have said something earlier)
I main janitor. As a janitor main, my greatest joy in life is slipping people who ignore my signs

I've seen some people complain about janitor borgs, so I decided to look into em

Unfortunately, not only is the janitor borg just a straight upgrade to janitors, it has absolutely no reason to use most of its kit
We give it standard cleaning supplies, and hell even bespoke tools to deal with leaving slippery tiles everywhere, but we also just let them clean anything they can walk over

This seems a bit too much to me, even for borgs. Also it's like, really boring

So what if we made their movement based cleaning cost something? How about we make it suck water from their bucket. Use the same pattern as mop code, make it twice as expensive though. Give it a slowdown, some sound cues, and an action button to trigger it all

## Why It's Good For The Game

We're left with a little engine of hate that can rev up to deal with large messes, or use its soap or mop to deal with smaller stuff, or to conserve water. There's tradeoffs instead of one size fits all solution, and something more unique then just a roundstart pimping ride upgrade (It's also shit, don't worry I'm coming for you too janitor mains)

(Slowdown is something close to 2.5 tiles a second, which feels roughly right to me. Video may seem slower, I'm coding on a cpu from 2007 with no graphics card, you'll have to forgive me)
 
https://user-images.githubusercontent.com/58055496/149652724-c16a63d6-8f9d-415a-bda5-2eb704157107.mp4

Oh and of course they can in theory splash anything, since the concept of a jannieborg puttering along making a hallway lubbed is very funny to me

I like this a lot better then lmao jannieborg go brr. Lemme know what you think yeah?

Oh and since they're going to need to actually use water now, I've traded out the borg's dense holosign projector for just a standard nondense one. Deploys instantly instead of having a 2 second do_after.

Ah and I threw some minor reagent code cleanup in too, to make my job a bit easier

(I can't tell if the blue stuff at the borg's feet is water or flaps, and I think they look significantly worse without it so I will choose to believe in flaps)
## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Added an action button to janitor cyborgs. It'll draw water from your bucket and clean the ground under you as you walk. More wasteful then mopping mind
del: Removed the janitor cyborg's autoclean feature
balance: Swapped out the janitor cyborg's dense holosign projector for a non dense version. It doesn't have the 2 second wait attached to it
refactor: Made being a cleaning chemical into a bitflag instead of having manual checks in mop code
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

LemonInTheDark committed:
Removes autowashing from janitor borgs

LemonInTheDark committed:
Moves "cleanable reagents" from a hardcoded typecheck to a reagent flag

LemonInTheDark committed:
Reworks janitor cyborg cleaning

I main janitor. As a janitor main, my greatest joy in life is slipping people who ignore my signs

I've seen some people complain about janitor borgs, so I decided to look into em

Unforunately, not only is the janitor borg just a straight upgrade to janitors, it has absolutely no reason to use most of its kit
We give it standard cleaning supplies, and hell even bespoke tools to deal with leaving slippery tiles everywhere, but we also just let them clean anything they can walk over

This seems a bit too much to me, even for borgs. Also it's like, really boring

So what if we made their movement based cleaning cost something? How about we make it suck water from their bucket. Use the same pattern as mop code, make it twice as expensive though. Give it a slowdown, some sound cues, and an action button to trigger it all

We're left with a little engine of hate that can rev up to deal with large messes, or use its soap or mop to deal with smaller stuff, or to conserve water. There's tradeoffs instead of one size fits all solution

Oh and of course they can in theory splash anything, since the concept of a jannieborg puttering along making a hallway lubbed is very funny to me

I like this a lot better then lmao jannieborg go brr. Lemme know what you think yeah?

LemonInTheDark committed:
Exchanges janitor borg's dense sign projector with a faster fully holographic one

carshalash commented:
This is a horrible idea dude.

JosephJomama commented:
Less brainless janiborg gameplay, more funny slipping gameplay, truly what is not to love

LemonInTheDark commented:
Rev audio is based on https://freesound.org/people/kvgarlic/sounds/157370/, it's CC 0 so I didn't want to clutter the credits file with it, but I may as well mention it here

Time-Green commented:
It's everything I could've hoped for. Finally I'll be able to roleplay those big scrubbing cleaning machines

OrionTheFox commented:
> It's everything I could've hoped for. Finally I'll be able to roleplay those big scrubbing cleaning machines

![image](https://user-images.githubusercontent.com/76465278/149619175-16f0ead4-b528-4e45-8f5d-b7a322df1191.png)

JohnFulpWillard commented:
I think this would be cool as an upgrade since you’re running around making entire hallways slippery

SuperNovaa41 commented:
this idea is better than mine

zxaber commented:
>and an action button to trigger it all

It should still be a module item.

The slowdown in the video seems horrible but I understand where you're going with it. How many tiles can a single bucket of water get you?

carshalash commented:
> > and an action button to trigger it all
> 
> It should still be a module item.
> 
> The slowdown in the video seems horrible but I understand where you're going with it. How many tiles can a single bucket of water get you?

24 tiles it seems from a full bucket of water. Just reopen the goof PR if you want to make it this bad.

Kubisopplay commented:
24 tiles of cleaning with major slowdown and long activation time. Just kill the borg, not leave it as barely usable zombie

theOOZ commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+	if(block_buffer_change)
+		return FALSE
+
+	var/obj/item/reagent_containers/glass/bucket/our_bucket = locate(/obj/item/reagent_containers/glass/bucket) in robot_owner.model.modules
```
It has to be a bucket? I can't use a larger container like a bluespace beaker later?

Fikou replied:
where will you get a bluespace beaker as a janiborg

theOOZ replied:
Printing them when available at R&D or whichever department lathe is closest by, is that not a thing here?

LemonInTheDark replied:
How are you gonna get a bluespace beaker into your equipment? is the idea to drag it around?

theOOZ replied:
sorry for the late reply! i figured the roboticist could swap out whatever reagent container is inside of the janiborg, along with what's in it sorta like deconstructing/upgrading a chemmaster.

LemonInTheDark replied:
I don't think that's currently possible, might be ok as a follow up pr. Not something I wanna do now tho

Ghommie commented:
I don't play cyborg and kind of dislike most of them, but some janitors really deserve the middle finger from robots for using the mop all the time or for the most part - I know, the janitor from Scrubs is a funny archetype, but at least he used floor signs -, when more efficient cleaning devices exist and getting them is not a trial - just mix water with ammonia -, so janiborgs, as brainless as they may be to play, get some sympathy from me in the current state.

I'm not calling this a bad PR but I have mixed feelings. I know from experience that  a stubborn janitor can be dealt with, even if temporarily, by stealing and spacing his janicart, mop, bucket, other stuff and even ID if he proves to be resilient. The same can't be done to an insufferable janiborg that's repeatedly making the floor slippery. I could drop a law 2 request but it'll probably stop caring about it after a short while or go full ham and disobey "because blood causes disease which is law 1" so I'm left with a conundrum: Do I risk overstepping the rules by sending it back to the roboticist as an MMI or do I turn the other cheek in spite of the fact it's probably pestering several other players too? Perhaps I'm exaggerating this small issue but it seems this PR is doing a good job in making them more detestable.

Time-Green commented:
I do think the startup is a bit extreme though

anconfuzedrock commented:
yeahh, I want to like this PR, but that looks pretty slow, slower than spraying space cleaner (granted that is limited) and with the windup, and that it's so limited, that might just be worse than using a mop. In that kind of windup you could clear a 3x3 area with the mop and a bit more, so it already has a big head start over that slow movement that clears one tile at a time. If you're cleaning one tile at a time with a mop as janitor, you're fucking stupid. At least gut the water thing, that's awkward.

StoneMason commented:
While it is a really heavy nerf to it, this will force janitor borgs to actually use the equipment that's given to them. They still have AA, making them better than their human counterparts, so I think this is pretty fair.

MetalClone commented:
Cyborgs are supposed to be supplemental to people on station, I wasn't here for security cyborgs being disabled but I suspect they were disabled for just being better than security officers. The janitor cyborg is literally just better than the janitor with the only downside being they have to stand still to recharge occasionally. Also giving them this kind of slip ability would make them more useful when hijacked by antagonists.

IndieanaJones commented:
Should have a bucket upgrade

bluetshirtguy commented:
What's with the sudden complaining for jannie borgs being specialists at cleaning, their job like you maintainers keep telling us, they seemed fine as before hand.

CuteTuret commented:
No one that is playing seriously will use that module, the water cost makes it so you can only clean a handful of tiles and you need to recharge the bucket again. 

Now here is something to consider, Engiborgs use energy for metal and glass, instead of carrying water in a bucket, why not make it so using the module or manual mopping has a moderate energy drain instead, that way having a bigger energy cell could equal to a "bigger bucket" upgrade. 

One last thing, borg upgrades tend to be very powerful, so maybe have an upgrade that removes some of the slowdown, maybe in the same node that unlocks the buffer upgrade for the pimpin ride?

carshalash commented:
> What's with the sudden complaining for jannie borgs being specialists at cleaning, their job like you maintainers keep telling us, they seemed fine as before hand.

Someone made an mspaint comic.

TemporalOroboros commented:
> (I can't tell if the blue stuff at the borg's feet is water or flaps, and I think they look significantly worse without it so I will choose to believe in flaps)

I think it's a set of the rotational mops that street sweepers and carwashes use.

ATH1909 commented:
maybe add an upgrade to make it normal speed?

and/or an expanded capacity bucket upgrade

LemonInTheDark commented:
> I think this would be cool as an upgrade since you’re running around making entire hallways slippery

I considered it, but I didn't want to take it away at roundstart

> It should still be a module item.
> 
> The slowdown in the video seems horrible but I understand where you're going with it. How many tiles can a single bucket of water get you?

I couldn't think of a good way for it to be an action item, was worried that it would feel disjointed 

You get 35 tiles for a full bucket.

>  Perhaps I'm exaggerating this small issue but it seems this PR is doing a good job in making them more detestable.

They could already do this, they just have little reason to. I'm not too worried about it

> yeahh, I want to like this PR, but that looks pretty slow, slower than spraying space cleaner (granted that is limited) and with the windup, and that it's so limited, that might just be worse than using a mop.

Good! I'm trying to hit a window where this is useful for large patches, and you'd want to use another tool for smaller ones. Need to do the math on what's exactly faster, thinking about tweaking water usage slightly.
I'm happy if it's a bit on the underpowered side, it's easy to buff, hard to nerf stuff in this codebase

> Should have a bucket upgrade

Maybe in future ye

> What's with the sudden complaining for jannie borgs being specialists at cleaning, their job like you maintainers keep telling us, they seemed fine as before hand.

I think it's sad to have all this gear we give janitor borgs and for them to have no reason to use it. Also like, it's really really strong, and I think mildly out of band for how much "better" we make borgs at some roles

> No one that is playing seriously will use that module, the water cost makes it so you can only clean a handful of tiles and you need to recharge the bucket again.
>
>Now here is something to consider, Engiborgs use energy for metal and glass, instead of carrying water in a bucket, why not make it so using the module or manual mopping has a moderate energy drain instead, that way having a bigger energy cell could equal to a "bigger bucket" upgrade.
>
>One last thing, borg upgrades tend to be very powerful, so maybe have an upgrade that removes some of the slowdown, maybe in the same node that unlocks the buffer upgrade for the pimpin ride?

I don't think the water cost is bad enough to make it prohibitive. I'm concerned at striking a midpoint between slipping all the hallways, and not being able to do anything. I'm ok with being conservative to start with (Video had a half empty bucket btw, which is why people are saying 24 tiles)

Don't want to autocreate water. I think that's kinda boring, and eliminates the use of soap

I don't want to decrease the slowdown too much. May be a follow up pr. Needs to be slow enough that mass slipping isn't too troublesome

> I think it's a set of the rotational mops that street sweepers and carwashes use.

YES

> maybe add an upgrade to make it normal speed?
> and/or an expanded capacity bucket upgrade

See above discussion

I'm gonna do some math on optimal wash times to make sure I'm hitting a good window here btw. (Never thought I'd ever say that) 
I'm not trying to make something that's as good as what exists now, I'm trying to hit a balance, and create different usecases. Jannieborgs don't have cleaning spray or grenades, the hope is this will fill a similar but different space

carshalash commented:
I don't get why everyone is so obsessed with making janiborg garbage entirely instead of just, making it so it needs an upgrade to autoclean like with the janicart. 

People are going to fucking hate janiborgs being near them if they're just loud slip machines.

LemonInTheDark commented:
> I don't get why everyone is so obsessed with making janiborg garbage entirely instead of just, making it so it needs an upgrade to autoclean like with the janicart.
> 
> People are going to fucking hate janiborgs being near them if they're just loud slip machines.

I don't see how the default janitor loadout - some consumables is garbage. My goal isn't to make this the only usable tool, it's to make it one of a few relatively sane solutions to problems. I don't think the noise is going to be an issue, I've specifically removed what tends to really bother people, high pitched droning. S just a low loud rumble combine with splash sounds.

Mothblocks commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+	wash_audio = new(owner)
+
+/datum/action/toggle_buffer/IsAvailable()
+	if(!istype(owner, /mob/living/silicon/robot))
```
Why not just `return istype`

LemonInTheDark replied:
Meant to use a ..() here, one sec

Mothblocks commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+
+	if(!buffer_on)
+		if(!COOLDOWN_FINISHED(src, activation_cooldown))
+			robot_owner.balloon_alert(robot_owner, "Auto-wash refreshing, please hold")
```
```suggestion
			robot_owner.balloon_alert(robot_owner, "auto-wash refreshing, please hold!")
```

All lowercase, `!` to signify failure

Mothblocks commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+		if(!allow_buffer_change())
+			return FALSE
+
+		robot_owner.balloon_alert(robot_owner, "Activating auto-wash")
```
```suggestion
		robot_owner.balloon_alert(robot_owner, "activating auto-wash...")
```

All lowercase, `...` to signify progress.

Mothblocks commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+			wash_audio.stop()
+			return FALSE
+	else
+		robot_owner.balloon_alert(robot_owner, "De-activating auto-wash")
```
```suggestion
		robot_owner.balloon_alert(robot_owner, "de-activated auto-wash")
```

Unless this also has a do_after, in which case `de-activating auto wash...`

Mothblocks commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+/datum/action/toggle_buffer/proc/allow_buffer_change()
+	var/mob/living/silicon/robot/robot_owner = owner
+	if(block_buffer_change)
+		robot_owner.balloon_alert(robot_owner, "Activation canceled")
```
```suggestion
		robot_owner.balloon_alert(robot_owner, "activation cancelled!")
```

Mothblocks commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+
+	var/obj/item/reagent_containers/glass/bucket/our_bucket = bucket_ref?.resolve()
+	if(!buffer_on && our_bucket?.reagents?.total_volume < 0.1)
+		robot_owner.balloon_alert(robot_owner, "Bucket is empty")
```
```suggestion
		robot_owner.balloon_alert(robot_owner, "bucket is empty!")
```

Mothblocks commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,167 @@
+	var/datum/reagents/reagents = our_bucket?.reagents
+
+	if(!reagents || reagents.total_volume < 0.1)
+		robot_owner.balloon_alert(robot_owner, "Bucket is empty, de-activating")
```
```suggestion
		robot_owner.balloon_alert(robot_owner, "bucket is empty, de-activating")
```

LemonInTheDark committed:
Lowers the movespeed penalty by 1 decisecond, brings the borg's move rate up to 2.5 tiles a second

LemonInTheDark committed:
Cleans up balloon alerts, properly uses a ..(), and adds starting/ending animations to match the audio cues

LemonInTheDark commented:
Increased speed from 2 to 2.5, added some extra animation and such. Updated the video to reflect this

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into jannie-orge

LemonInTheDark committed:
There you go linters

GuillaumePrata commented:
While I love the idea and it is 100% better than the free cleaning we have, trying to balance it around the water bucket will be pointless because a borg can drag a water tank and refill their bucket on the spot.

Shitter borgs will go around spam slipping people, balancing it around the speed makes killing them easier since it is needed to walk/crawl to fight them.

LemonInTheDark commented:
That's true, applies to janitor too now that I think of it, you could always just drag the thing behind you all the time. Seems like it'd be easy enough to just grab it and run tho.
S part of why the speed decrease is useful you're right. Maybe I should add a delay to getting out of the wash state to cut back on dumb usage a bit?

GuillaumePrata commented:
A "stun" before going back to full speed would stop a lot of the annoyance in dealing with borgs that do this, since they can make slip pools and while you slow down to try and approach them, they run.

I am overthinking this but I know a lot of players just endure/ignore shitter Silicon players because they think silicon combat takes too much effort while a borg can just spam the sniffer at robo and respawn.

LemonInTheDark commented:
> A "stun" before going back to full speed would stop a lot of the annoyance in dealing with borgs that do this, since they can make slip pools and while you slow down to try and approach them, they run.

I'm good with this, if only because it makes the effect feel nicer. May be too long of a delay in total, will see how things work out. I can always recut the startup sound to fit a new delay.

> I am overthinking this but I know a lot of players just endure/ignore shitter Silicon players because they think silicon combat takes too much effort while a borg can just spam the sniffer at robo and respawn.

This is an admin issue. Ban them.

LemonInTheDark committed:
Makes the startup sound mesh better with the others
Adds a ending delay to the movement slowdown, length of the end audio + the remaining time on the current clip.
Hopefully this will prevent sillyness
Refactored the code a bit to make it at least slightly less messy, sorry bout this

LemonInTheDark commented:
I've added an ending delay of 2.6 + (some time between 2 seconds and 0) to allow for audio to fully finish, and to prevent silly borg plays. Video's been updated too

EricZilla commented:
I like that it lets you choose between using your drying agent or just letting everyone suffer.

(Why can borgs make tequila but not space cleaner? Its a mystery to all of us)

carshalash commented:
I REALLY don't get why you guys can't just make it so that janiborgs need the buffer mod to clean instead of making them a loud slipping grief module.

Bawhoppen commented:
The blue stuff on the bottom always looked something like this to me:
![image](https://user-images.githubusercontent.com/6510293/149658665-2d79a8d0-9edb-4c3e-8568-d2843b6a688e.png)
![image](https://user-images.githubusercontent.com/6510293/149658688-5bba56a7-3046-4106-b513-8f0b3716d6bb.png)
![image](https://user-images.githubusercontent.com/6510293/149658697-f05ccd65-f85c-43f4-9a17-84aba937b888.png)

SuperNovaa41 commented:
> I REALLY don't get why you guys can't just make it so that janiborgs need the buffer mod to clean instead of making them a loud slipping grief module.

because its funny this way

LemonInTheDark commented:
> I REALLY don't get why you guys can't just make it so that janiborgs need the buffer mod to clean instead of making them a loud slipping grief module.

I've explained why I think the buffer mod is bad, and what my goal here quite thoroughly I thought. You and I just seem to disagree on base principles

Dorsisdwarf commented:
I'll be honest I don't understand the idea some people promote that silicons now have to be _worse_ at their jobs than the equivalent human job? 

Like, isn't the entire point of borgs that they have laserlike focus and are generally useless outside their lane (Unlike the real janitor who can go "nukies" and pick up a lasgun, isn't bound by laws/forced to do whatever asinine tasks the crew are demanding etc), and in exchange they get to be top-tier at their jobs?

Also afaik secborgs weren't cut for being OP, they were cut because people consistently and persistently refused to play them like cyborgs instead of robocops

Dorsisdwarf commented:
That said the way the borg jiggles around is extremely funny

LemonInTheDark commented:
> I'll be honest I don't understand the idea some people promote that silicons now have to be _worse_ at their jobs than the equivalent human job?

What do you mean by worse? I don't think they should be a budget janitor, I just think it's sad how *much* better they are, and how the way that they're better basically outmodes a lot of the appeal of janitor, alongside all of their kit.

AA is genuinely pretty powerful, even outside having a unique tool still. If I wanted them to be at about the same level I'd just cut the buffer upgrade. They wouldn't have much mass cleaning power, but they could AA into all the rooms on the station.

> Also afaik secborgs weren't cut for being OP, they were cut because people consistently and persistently refused to play them like cyborgs instead of robocops

Yeah idk why people were talking about secborgs being op, they got disabled because ai laws + sec escalation is hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh on all sides

CuteTuret commented:
> I'll be honest I don't understand the idea some people promote that silicons now have to be _worse_ at their jobs than the equivalent human job?
> 
> Like, isn't the entire point of borgs that they have laserlike focus and are generally useless outside their lane (Unlike the real janitor who can go "nukies!!" and pick up a lasgun, isn't bound by laws/forced to do whatever asinine tasks the crew are demanding etc), and in exchange they get to be top-tier at their jobs?
> 
> Also afaik secborgs weren't cut for being OP, they were cut because people consistently and persistently refused to play them like cyborgs instead of robocops

Yknow it doesn't help the coder starts his PR with "I main janitor, I love slipping people, janiborgs do my job while I am otherwise busy griefing, so I'll make them worse at their job".

Dorsisdwarf commented:
> > I'll be honest I don't understand the idea some people promote that silicons now have to be _worse_ at their jobs than the equivalent human job?
> > Like, isn't the entire point of borgs that they have laserlike focus and are generally useless outside their lane (Unlike the real janitor who can go "nukies!!" and pick up a lasgun, isn't bound by laws/forced to do whatever asinine tasks the crew are demanding etc), and in exchange they get to be top-tier at their jobs?
> > Also afaik secborgs weren't cut for being OP, they were cut because people consistently and persistently refused to play them like cyborgs instead of robocops
> 
> Yknow it doesn't help the coder starts his PR with "I main janitor, I love slipping people, janiborgs do my job while I am otherwise busy griefing, so I'll make them worse at their job".

Dont forget forcing them to grief in order to do their job so that the crew borgtips them every time

EricZilla commented:
> Dont forget forcing them to grief in order to do their job so that the crew borgtips them every time

Just fill the bucket with space cleaner instead of water, or like, use your signs, or your drying spray

carshalash commented:
> > Dont forget forcing them to grief in order to do their job so that the crew borgtips them every time
> 
> Just fill the bucket with space cleaner instead of water, or like, use your signs, or your drying spray

I'm pretty sure you can't use your bucket on chem dispensers.

SuperNovaa41 commented:
> > > Dont forget forcing them to grief in order to do their job so that the crew borgtips them every time
> > 
> > 
> > Just fill the bucket with space cleaner instead of water, or like, use your signs, or your drying spray
> 
> I'm pretty sure you can't use your bucket on chem dispensers.

you absolutely can

zxaber commented:
Janiborg buckets are not held in an apparatus like mediborg beakers. They're given nodrop because the borg is not supposed to be able to let go.

cacogen commented:
This is such a braindead PR. Anybody who has played janitor long enough knows it's slow, tedious and impossible to keep up with the mess on the station. The floor buffer upgrade for the janicart was a blessing. Everybody should clean as fast as the janiborg does. It's fun, easy and satisfying. Unlike picking up trash and replacing lights, which is another responsibility people forget the janitor has on top of cleaning floors that's impossible to keep up with.

The floor buffer and vacuum upgrades for the janicart shouldn't be mutually exclusive. The janitorial chem sprayer should be unlockable for regular janitors (instead of just ERT, who when spawned steal all the janitor's glory) as well. And there should be a faster way of picking up trash/sorting trash from stuff simply small enough to fit in the bag and replacing lights as well.

TheSmallBlue commented:
Whenever I go cyborg, the module I use the most is the janitorial one, so I have a few opinions.
I do agree that there's something wrong with the janiborg as it is right now, though its hard to say what. 
Don't think that the problem is that he's way too overpowered, since what I've seen is that there's a point in the round in which all janitors go SSD and only the janiborg is able to clean the extreme amount of mess. This happens in every. single. round.
However, when there is an active janitor, it is true that the janiborg will fucking overpower them in every single way, making their job utterly useless.

I do like some of the ideas in this PR, like the whole automatic-cleaning-needing-to-have-a-cleaning-substance-stored thing. Dunno about the revving up though, I'd like to test it in a PR or something. Maybe a better idea would be to link Janiborg availability with the available janitorial slots. Like, if there's two occupied janitor slots out of two and one of the janitors disconnects, a borg gains the availability to change into a janiborg.

Also since this is a "rework", can you PLEASE make it so that the janiborg can drag things into the trash chute or tables? I can't even begin to talk about how much I hate those times when I see a pile of trash in the floor and I use the bluespace trashbag, only for it to pick up NOTHING because the pile consists of dismembered arms.

LemonInTheDark commented:
> I do like some of the ideas in this PR, like the whole automatic-cleaning-needing-to-have-a-cleaning-substance-stored thing. Dunno about the revving up though, I'd like to test it in a PR or something. Maybe a better idea would be to link Janiborg availability with the available janitorial slots. Like, if there's two occupied janitor slots out of two and one of the janitors disconnects, a borg gains the availability to change into a janiborg.

I don't think we need to restrict the role that much, but I guess we'll see. 
The delays here may need changing in future, I'm intentionally highballing them to encourage people to use more then their existing strategy for cleaning grime

> Also since this is a "rework", can you PLEASE make it so that the janiborg can drag things into the trash chute or tables? I can't even begin to talk about how much I hate those times when I see a pile of trash in the floor and I use the bluespace trashbag, only for it to pick up NOTHING because the pile consists of dismembered arms.

What exactly do you mean?

TheSmallBlue commented:
> What exactly do you mean?

It's not really related to the changes you've made in this PR, but it is a pet peeve I have when playing Janiborg. The only way the Janiborg can clean items is by stuffing them into their trash bag, then using that trash bag on a trash bin. Thing is, some items are too big to fit into the trashbag, like body parts or clothes. So if you're cleaning medical and you want to get rid of the bloody body parts stinking up the otherwise nice and clean floor, you can drag them around, sure, but you can't place them inside the trash. So you have no choice but to leave them on the floor.

LemonInTheDark commented:
> > What exactly do you mean?
> 
> It's not really related to the changes you've made in this PR, but it is a pet peeve I have when playing Janiborg. The only way the Janiborg can clean items is by stuffing them into their trash bag, then using that trash bag on a trash bin. Thing is, some items are too big to fit into the trashbag, like body parts or clothes. So if you're cleaning medical and you want to get rid of the bloody body parts stinking up the otherwise nice and clean floor, you can drag them around, sure, but you can't place them inside the trash. So you have no choice but to leave them on the floor.

The broom upgrade would be useful for this wouldn't it?  (I think the bodyparts thing is mostly heads and such, pretty sure limbs fit in trashbags just fine)

MrMelbert commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,205 @@
+
+/datum/action/toggle_buffer
+	name = "Activate Auto-Wash"
+	desc = "Trade speed and water for a clean floor"
```
```suggestion
	desc = "Trade speed and water for a clean floor."
```

LemonInTheDark replied:
Finally, we've come full circle

MrMelbert commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -350,14 +348,205 @@
+	robot_owner.remove_movespeed_modifier(/datum/movespeed_modifier/auto_wash)
+	UpdateButtonIcon()
+
+/datum/action/toggle_buffer/proc/allow_buffer_change()
```
Some of these procs should be documented

LemonInTheDark replied:
Sure

LemonInTheDark committed:
Adds a period to make a sentence well, a sentence

The revenge of the bugs

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

LemonInTheDark committed:
Renames allow_buffer_change to allow_buffer_activate, properly autodocs all the unique procs

LemonInTheDark committed:
Merge branch 'jannie-orge' of https://github.com/LemonInTheDark/tgstation into jannie-orge

CameronWoof commented:
> This doesn't feel like a good change at all, you're taking away the janiborgs 1 job, which is to clean, and replacing it with griefing because that's what you like to do as a normal janitor. AA really isn't a buff in any sense, if people want somewhere to be cleaned they can let you in, or hack your way in if it's that important.
> 
> It's not like the borgs are even "stealing" your work either, if you want to clean, law 2 the borg to pick another module. 'Blood causes diseases therefore law 1' sounds closer to them trying to circumvent having to obey law 2 than actually wanting to clean, in which case you should take it up with an admin.

Silicon policy protects against ordering a borg to change their module outside of a good reason (and "because we both showed up to work in the same color outfit" isn't a good reason)

GuillaumePrata commented:
>Borgs are supposed to be really especially good at their jobs. It's the upside to the whole "shackled by laws and limited to one module with specific things in it instead of hands" thing. I'm not sure why everyone woke up like a week ago, realized borgs were good at the jobs they're specialized in

A lot of people are ignoring the point that this version of the Janiborg is just more interesting for everyone involved than the braindead auto cleaning it had before.

We can talk balance, grief potential or what niche borgs should have all day, nothing will change that this version is more interesting than the previous one.
IMO we should push borgs to be more like this, fill the same job as a human but with their unique spins on how it is done so the difference between a borg and a human is more than AA + Spaceproof - hands.

LemonInTheDark commented:
> The dictionary definition of 'interesting' does unfortunately afford that the loud slow floor-wetting device is more interesting than one that just cleans without being loud, slowing you down, or wetting the floor. However, "more interesting" does not mean "better" or "more fun". As it stands, I can't actually imagine this new thing seeing use at all, even if the vaunted math bears out that it's slightly faster than using a mop.
> 
> This is just a flat nerf to the module (replacing something good with something _interesting_), but I've seen no compelling reason the scrubber upgrade can't simply be delayed until both the borg and the janitor have access to it.

Of course it's a nerf, I think the current module is so good that it makes almost all of the borg's tools and upgrades totally useless. I don't like that, so I'm replacing it. If it doesn't see use I'm open to changing aspects of it, but I'm starting here because buffing is something feature coders will do on their own, but nerfing is very much not.

I don't want to just delay the auto-clean behavior to upgrades because A: We'd still have an upgrade that makes the other upgrades/most all of the borg's kit totally useless, and B: I don't like the idea of making jannieborgs just humans with no grenades or space cleaner at roundstart. I think borgs should be unique and slightly better then their human counterparts, rather then slightly less good, or worse carbon copies.

> This doesn't feel like a good change at all, you're taking away the janiborgs 1 job, which is to clean, and replacing it with griefing because that's what you like to do as a normal janitor. AA really isn't a buff in any sense, if people want somewhere to be cleaned they can let you in, or hack your way in if it's that important.
> 
> It's not like the borgs are even "stealing" your work either, if you want to clean, law 2 the borg to pick another module. 'Blood causes diseases therefore law 1' sounds closer to them trying to circumvent having to obey law 2 than actually wanting to clean, in which case you should take it up with an admin.

I think you're arguing with john's comics more then you're arguing with me. 
I don't think removing the autoclean somehow makes janiborgs useless at cleaning, or takes their job away. They still have a good few tools to clean with, more with upgrades.

AA is totally a buff, seems insane to even mention it. Not sure how big of one it is though, which is why I'm not just totally nuking the autoclean.

Honestly I mentioned the slipping somewhat facetiously. I find watching people running through the halls and slipping on my clearly marked wetspots funny yes, but we even give jannieborgs tools to prevent that if they want. You wouldn't know that because right now there's absolutely no reason to use it, but they have spray that drys puddles.

The rest of this is I think arguing with a strawman more then anything. Borgs can't steal my work if I don't play the game.

Timberpoes reviewed (approved):
Too long; didn't review

cacogen commented:
This is such a miserable change. 

What used to be fast, fun and satisfying is now tedious, agonising and boring. It takes so long to activate the cleaning mode, it cleans barely anything before your bucket is empty, it uses a shitload of water and then it takes time to deactivate as well. 

Meanwhile everybody's complaining because of all the wet floors you're leaving in your wake (and placing signs is both needless and makes things even more tedious). 

Completely unnecessary change when there are so many dirty tiles to clean across the station, you also have trash and replacing lights to juggle and you have to stop to mop or use soap anytime a mess is underneath something. 

Less ruinous changes to longstanding game mechanics by people who barely play the game please. I don't see the point in picking janiborg anymore.

TheSilverNuke commented:
Janiborg is a useless module now and I will never get the same enjoyment out of playing it. Better to go engiborg, make a bunch of water, and then spill the canisters everywhere. That is the new way to clean. Congratulations on ruining a good module.

TheSilverNuke commented:
> This is such a braindead PR. Anybody who has played janitor long enough knows it's slow, tedious and impossible to keep up with the rate of mess on the station on rounds above a certain pop. The floor buffer upgrade for the janicart was a blessing because of this. Everybody should clean as fast as the janiborg does. It's fun, easy and satisfying. You still have to stop to mop underneath objects, pick up and sort trash from regular objects and replace lights, which are all tedious jobs that are impossible to stay on top of.

>The floor buffer and vacuum upgrades for the janicart shouldn't be mutually exclusive. The janitorial chem sprayer should be unlockable for regular janitors (instead of just ERT, who steal all the janitor's glory). There should be a faster way of picking up/sorting trash and replacing lights as well. It's never been a fast enough job to keep up with the rate of mess outside of lowpop shifts (and even then you have plenty to deal with by default).

You have no idea how much this rings true to me. Even with a janiborg module I always found it difficult to keep up with the messes of a 50< pop crew. Blood absolutely everywhere, trash on the ground, broken lightbulbs. Janiborgs with unupgraded chargers barely even get enough lightbulb replacements to replace half a dozen lights at most over the course of 10 minutes, so you have to nag robotics for upgrades if you want to help fix the lights. Meanwhile janitors just grab a box and go or print a new light replacer. This PR hurts janiborgs to an unreasonable degree and I can't express my discontent enough.

TheSilverNuke commented:
Also stealth slaughter demon buff

bruhlookatthisdood commented:
so....
cyborg bucket expansion upgrade when?
let me hold more water ):<

cacogen commented:
It might be nice but becomes less desirable when you realise you can just drag a water tank and constantly refill. 

I think what hamstrings it the most other than how long tiles stay wet is the activation time.

adamkad1 commented:
Now it just cancels at the start of action

LemonInTheDark commented:
why are you bug reporting on my year old pr I can't do anything with that man

bruhlookatthisdood commented:
fix it

LemonInTheDark commented:
what the fuck even is the bug
